Quantum-Control-Applications-QuAM/Superconducting/README.md

```diff
@@ -0,0 +1,357 @@
+connectivity.add_qubit_flux_lines(qubits=qubits)
+connectivity.add_qubit_drive_lines(qubits=qubits)
+allocate_wiring(connectivity, instruments)
+```
+3. Finally, the wiring and network information is serialized and store in wiring.json and the QUAM state is initiated in state.json.
+The wiring and port mapping can also be visualized in a matplotlib figure. 
+
+```python
+from quam_libs.quam_builder.machine import build_quam_wiring
+from qualang_tools.wirer import visualize
+# Build the wiring and network into a QuAM machine and save it as "wiring.json"
+build_quam_wiring(connectivity, host_ip, cluster_name, path)
+
+# View wiring schematic
+visualize(connectivity.elements, available_channels=instruments.available_channels)
+```
+![opx1000_wiring](./opx1000_wiring.PNG)
+
+### [2. The QuAM components](./quam_libs/components)
+Describe the structure of the QuAM components and how they can be customized, what to pay attention to...
+Also, how to create a new one and update the wiring and build_quam accordingly.
+
+The hierarchy and structure of QUAM can be detailed as follows: 
+1. [quam_root.py](./quam_libs/components/quam_root.py) represents the highest level in terms of hierarchy.
+It contains the qubits and qubit pairs objects, as well as the wiring, network and Octaves. 
+Its methods are usually applied to all qubits or active qubits. 
+2. The definition of a QuAM transmon is defined in [transmon.py](./quam_libs/components/transmon.py). It contains the
+genral transmon attributes (T1, T2, f_01...) as well as the QuAM components composing the transmon (``xy``, ``resonator`` and 
+`z` in this case). Two-qubit gates can also be implemented by defining a specific qubit pair component as shown in 
+[transmon_pair.py](./quam_libs/components/transmon_pair.py).
+3. The sub-QuAM components are either defined from the base QUAM components directly, such as the qubit xy drive which is
+directly defined as an `IQChannel`, or from user-defined components such as 
+[readout_resonator](./quam_libs/components/readout_resonator.py) or [flux_line](./quam_libs/components/flux_line.py), 
+which allows the customization of their attributes.
+
+
+### [3. Generating the QuAM and state.json](./configuration/make_quam.py)
+Once the QuAM root is implemented with the subsequent components, the QuAM state can be generated automatically and each 
+parameter of the QUAM components is initialized to its arbitrary default value. 
+All of these parameters can be updated programmatically based on the specs from the chip manufacturer for instance and 
+the process is described in the next section.
+
+The script called [make_quam.py](./configuration/make_quam.py) takes care of generating the QUAM and filling the state 
+according to the wiring file. The Octaves connection parameters can also be edited here if relevant.
+
+Everything happens in the `build_quam` function, which programmatically adds all the Octaves, ports, transmons and pulses
```

Comment:
Replace `Everything happens` -> `The QuAM generation happens` -- to be more accurate and explanatory

Reply:
Done